### PR TITLE
Fix data rereads

### PR DESCRIPTION
**Pull request by cbowman0**
Cherry-picked the commits to fix issue #1521 against the master branch.

My peers were experiencing this exact problem yesterday.  We are testing this fix now.

### CODE REVIEW

File: webapp/graphite/render/functions.py [hunk 1]
```diff
@@ -2328,24 +2290,45 @@ def holtWintersForecast(requestContext, seriesList):
...
+    data = analysis['predictions']
+    windowPoints = previewSeconds / data.step
+    forecast = TimeSeries(data.name, data.start + previewSeconds, data.end, data.step, data[windowPoints:])
+    forecast.pathExpression = data.pathExpression
+
+    data = analysis['deviations']
+    windowPoints = previewSeconds / data.step
+    deviation = TimeSeries(data.name, data.start + previewSeconds, data.end, data.step, data[windowPoints:])
+    deviation.pathExpression = data.pathExpression
+
```
Comment (JeanFred) [inline webapp/graphite/render/functions.py]: This makes Travis lint check fail:
```
graphite-web/webapp/graphite/render/functions.py:2331:1: W293 blank line contains whitespace
```

**Comment by obfuscurity** [comment]
@cbowman0 How well has this been working for you in production (against master)?

**Comment by cbowman0** [comment]
I will attempt to get test coverage on these once I work out how to simulate the request object.  If someone has any ideas, please let me know.

**Comment by cbowman0** [comment]
@obfuscurity I'll let you know early next week.  I deployed the changes to the primary cluster today.

**Comment by cbowman0** [comment]
No complaints from my users.

**Comment by cbowman0** [comment]
Any objections to merging this?

**Comment by obfuscurity** [comment]
@cbowman0 I haven't had a time to review this yet. Sorry, been hands-full with $newjob and Monitorama. Will try to look closer while I'm traveling tomorrow.

**Comment by obfuscurity** [comment]
@cbowman0 Sorry for the delay. Been HAM on Monitorama lately. This looks sane and we already merged a version of this in 0.9.x, so there's no reason not to get this in master and let people hammer away at it. Thanks for this fix.

**Pull request merged**